rgdps/api/gd/user_relationships.py

```diff
@@ -0,0 +1,282 @@
+        ctx,
+        sender_user_id=target_id,
+        recipient_user_id=user.id,
+        request_id=request_id,
+    )
+
+    if isinstance(result, ServiceError):
+        logger.info(
+            f"Failed to accept friend request from {target_id} with error {result!r}.",
+        )
+        return responses.fail()
+
+    logger.info(f"{user} successfully accepted friend request from {target_id}.")
+    return responses.success()
+
+
+async def user_relatoionship_list_get(
+    ctx: HTTPContext = Depends(),
+    user: User = Depends(authenticate_dependency()),
+    relationship_type: UserRelationshipType = Form(..., alias="type"),
+):
+
+    result = await user_relationships.get_user(
+        ctx,
+        user.id,
+        relationship_type,
+    )
+
+    if isinstance(result, ServiceError):
+        logger.info(
+            f"Failed to view user relationship of {user.id} with error {result!r}.",
+        )
+        return responses.fail()
+
+    if not result.relationships:  # No relationships found.
+        return responses.code(-2)
+
+    response = "|".join(
+        gd_obj.dumps(
+            [
+                gd_obj.create_profile(relationship.user),
+                gd_obj.create_user_relationship_list(relationship.relationship),
+            ],
+        )
+        for relationship in result.relationships
+    )
+
+    # Mark them as seen.
+    await user_relationships.mark_all_as_seen(ctx, user.id, relationship_type)
+    logger.info(f"{user} successfully viewed user relationship.")
+
+    return response
+
+
+async def user_friend_remove(
+    ctx: HTTPContext = Depends(),
+    user: User = Depends(authenticate_dependency()),
+    target_id: int = Form(..., alias="targetAccountID"),
+):
+    result = await user_relationships.remove_friendship(
+        ctx,
+        user.id,
+        target_id,
+    )
+
+    if isinstance(result, ServiceError):
+        logger.info(
+            f"Failed to remove friend {target_id} with error {result!r}.",
+        )
+        return responses.fail()
+
+    logger.info(f"{user} successfully removed friend {target_id}.")
+    return responses.success()
+
+
+async def block_user(
+    ctx: HTTPContext = Depends(),
+    user: User = Depends(authenticate_dependency()),
+    target_id: int = Form(..., alias="targetAccountID"),
+):
+
+    # Remove friendship if exists.
+    friendship = await user_relationships.get_user(
+        ctx,
+        user.id,
+        UserRelationshipType.FRIEND,
+    )
+
+    if not isinstance(friendship, ServiceError):
+        await user_relationships.remove_friendship(
+            ctx,
+            user.id,
+            target_id,
+        )
+
+    result = await user_relationships.create(
+        ctx,
+        user.id,
+        target_id,
+        UserRelationshipType.BLOCKED,
+    )
+
+    if isinstance(result, ServiceError):
+        logger.info(
+            f"Failed to block user {target_id} with error {result!r}.",
+        )
```

Comment:
Can you not `return responses.fail()` here?

Reply:
Even if it makes no difference, would make the code clearer.

Reply:
this endpoint returns `1` no matter if the request failed

Reply:
I'd say to still do it regardless as long as it doesnt break anything. Makes reading the code clearer.

Reply:
![image](https://github.com/RealistikDash/RealistikGDPS/assets/64559993/ea52e790-5ee4-42ec-8cae-eb53833f64e5)

Seems the client has the reserved handler for when it hits `responses.fail()`, added in 9efe9c5